integration_tests/src/main/python/orc_cast_string_test.py

```diff
@@ -24,7 +24,7 @@ def test_casting_string_to_integers(spark_tmp_path, to_type):
     orc_path = spark_tmp_path + '/orc_cast_string_to_int'
     normal_cases = []
     length = 2048
-    while len(normal_cases) < length:
+    for _ in range(0, length):
         normal_cases.append(str(random.randint(INT_MIN, INT_MAX)))
```

Comment:
Sorry, but NO. This test is not reproducible.  You are not setting a seed for random at all, which means you will get different data each time the test runs. So if there is a bug in here we will not be able to debug it if the test fails.